vscode/microsoft-kiota/src/generateClient.ts

```diff
@@ -24,35 +24,31 @@ export function generateClient(context: vscode.ExtensionContext,
   operation: ConsumerOperation,
   workingDirectory: string = getWorkspaceJsonDirectory()
 ): Promise<KiotaLogEntry[] | undefined> {
-  try {
-    return connectToKiota<KiotaLogEntry[]>(context, async (connection) => {
-      const request = new rpc.RequestType1<GenerationConfiguration, KiotaLogEntry[], void>(
-        "Generate"
-      );
-      return await connection.sendRequest(
-        request,
-        {
-          cleanOutput: cleanOutput,
-          clearCache: clearCache,
-          clientClassName: clientClassName,
-          clientNamespaceName: clientNamespaceName,
-          deserializers: deserializers,
-          disabledValidationRules: disableValidationRules,
-          excludeBackwardCompatible: excludeBackwardCompatible,
-          excludePatterns: excludeFilters,
-          includeAdditionalData: includeAdditionalData,
-          includePatterns: includeFilters,
-          language: language,
-          openAPIFilePath: descriptionPath,
-          outputPath: output,
-          serializers: serializers,
-          structuredMimeTypes: structuredMimeTypes,
-          usesBackingStore: usesBackingStore,
-          operation: operation
-        } as GenerationConfiguration,
-      );
-    }, workingDirectory);
-  } catch (error) {
-    return Promise.resolve(undefined);
-  }
+  return connectToKiota<KiotaLogEntry[]>(context, async (connection) => {
+    const request = new rpc.RequestType1<GenerationConfiguration, KiotaLogEntry[], void>(
+      "Generate"
+    );
+    return await connection.sendRequest(
```

Comment:
But we are awaiting the connection.sendRequest which is very likely to yield an error if the request fails. If unhandled, this would cause the program to crash/become unstable. Maybe we should await the creation of connection too to make this complete but am not sure removing them is a good idea. What do you think @thewahome , @calebkiage ? Do we remove and handle it better in #5438

Reply:
Agree, no last-minute changes today please.

Reply:
@fey101, I think in that case, the try catch should be inside the callback of the connection creation. What Vincent is saying is that the promise creation never fails so the try catch around it will never execute the catch code path.

Reply:
thank you for articulating the technical aspects better than I originally could. yes, what I'm trying to outline here is that we'll NEVER go through the catch block as is because of the lack of await.
And if we take a step back, I'm trying to outline the fact that if those try catch were added to "improve reliability", they are NOT achieving that in the current form.